psydac/linalg/kron.py

```diff
@@ -27,73 +29,71 @@ def kron_dot(starts, ends, pads, X, X_tmp, Y, A, B):
         for i2 in range(s2, e2+1):
              Y[i1-s1+p1,i2-s2+p2] = sum(A[i1, k]*X_tmp[i1-s1+k, i2-s2+p2] for k in range(2*p1+1))
     return Y
+
 #==============================================================================
-class KroneckerStencilMatrix( LinearOperator ):
+class KroneckerStencilMatrix_2D( Matrix ):
 
-    def __init__( self,V, W, *args ):
+    def __init__( self, V, W, A1, A2 ):
 
         assert isinstance( V, StencilVectorSpace )
         assert isinstance( W, StencilVectorSpace )
+        assert V is W
+        assert V.ndim == 2
 
-        for i,A in enumerate(args):
-            assert isinstance( A, StencilMatrix )
-            assert A.domain.ndim == 1
-            assert A.domain.npts[0] == V.npts[i]
+        assert isinstance( A1, StencilMatrix )
+        assert A1.domain.ndim == 1
+        assert A1.domain.npts[0] == V.npts[0]
 
+        assert isinstance( A2, StencilMatrix )
+        assert A2.domain.ndim == 1
+        assert A2.domain.npts[0] == V.npts[1]
+
+        self._space = V
+        self._A1    = A1
+        self._A2    = A2
+        self._w     = StencilVector( V )
 
-        self._domain   = V
-        self._codomain = W
-        self._mats     = args
     #--------------------------------------
     # Abstract interface
     #--------------------------------------
     @property
     def domain( self ):
-        return self._domain
+        return self._space
 
     # ...
     @property
     def codomain( self ):
-        return self._codomain
-
-    # ...
-    @property
-    def mats(self):
-        return self._mats
+        return self._space
```

Comment:
This new implementation assumes that domain and codomain are the same. This looks totally wrong to me.

Reply:
okay I don't remember why was this changed lol

Reply:
in anyway, we have an assert that the domain is equal to the codomain

Reply:
I think that assert is new, it should not be there

Reply:
okay this class is for 2d stencil matrix, it's not used at all, so it doen't really matter, we are using the general one, we just need to keep this one for the moment until we improve the general, and then we can delet it

Reply:
OK. Please comment it out so that it's clear that we are not using it. Or maybe put it in a separate file to be removed later.

Reply:
I will comment it out and when we don't need it, we can delete it